src/components/coblocks-deactivate-modal/modal.js

```diff
@@ -0,0 +1,184 @@
+			if ( !! choice.text_field ) {
+				textFields[ choice.text_field ] = '';
+			}
+		} );
+		setFormData( {
+			choices: [],
+			coblocks_version: coblocksDeactivateData.coblocksVersion,
+			domain: coblocksDeactivateData.domain,
+			hostname: coblocksDeactivateData.hostname,
+			language,
+			persona: coblocksDeactivateData.wpOptions?.persona,
+			skill: coblocksDeactivateData.wpOptions?.skill,
+			wp_version: coblocksDeactivateData.wpVersion,
+			...textFields,
+		} );
+	};
+
+	const onDeactivate = async ( submit = false ) => {
+		if ( submit && formData.choices.length >= feedbackData.choices_min ) {
+			await fetch( `${ API_BASE_URL }/feedback/coblocks-optout`, {
+				method: 'POST',
+				body: JSON.stringify( formData ),
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			} );
+		}
+		setOpen( false );
+		window.location.href = href;
+	};
+
+	const onCheckboxChange = ( isChecked, slug ) => {
+		setFormData( ( prevFormData ) => {
+			const choices = prevFormData.choices;
+			if ( isChecked ) {
+				choices.push( slug );
+			} else {
+				choices.splice( choices.indexOf( slug ), 1 );
+			}
+			return {
+				...prevFormData,
+				choices,
+			};
+		} );
+	};
+
+	const onTextChange = ( key, value ) => {
+		setFormData( ( prevFormData ) => ( {
+			...prevFormData,
+			[ key ]: value,
+		} ) );
+	};
+
+	if ( ! isOpen || ! feedbackData ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className="coblocks-plugin-deactivate-modal"
+			onRequestClose={ () => setOpen( false ) }
+			title={ feedbackData.labels.title }
+		>
+			<div className="coblocks-plugin-deactivate-modal__checkbox">
+				{ feedbackData.choices.map( ( choice ) => {
+					const isChecked = formData.choices.indexOf( choice.slug ) >= 0;
+					return (
+						<div key={ choice.slug }>
+							<CheckboxControl
+								checked={ isChecked }
+								label={ choice.label }
+								onChange={ ( checked ) => onCheckboxChange( checked, choice.slug ) }
+							/>
+							{ !! choice.text_field && (
+								<TextControl
+									className={ `coblocks-plugin-deactivate-modal__text ${
+										isChecked ? 'show' : ''
+									}` }
+									onChange={ ( value ) => onTextChange( choice.text_field, value ) }
+									value={ formData[ choice.text_field ] }
+								/>
+							) }
+						</div>
+					);
+				} ) }
+			</div>
+			<ButtonGroup>
+				<Button
+					className="coblocks-plugin-deactivate-modal__button"
+					onClick={ () => onDeactivate( true ) }
+					variant="primary"
+				>
+					{ feedbackData.labels.submit_deactivate }
+				</Button>
+				<Button
+					className="coblocks-plugin-deactivate-modal__button"
+					onClick={ () => onDeactivate( false ) }
+					variant="link"
+				>
+					{ feedbackData.labels.skip_deactivate }
+				</Button>
+			</ButtonGroup>
+
+			<footer className="coblocks-plugin-deactivate-modal__footer">
+				<div
+					dangerouslySetInnerHTML={ {
+						__html: feedbackData.labels.privacy_disclaimer,
```

Comment:
For added safety here I would wrap around this method https://github.com/WordPress/gutenberg/tree/07051aa1caab9e02f11b15ff7972a5ad3aff15ae/packages/dom#safehtml. See usage here: https://github.com/WordPress/gutenberg/blob/07051aa1caab9e02f11b15ff7972a5ad3aff15ae/packages/edit-widgets/src/components/sidebar/widget-areas.js#L57